cli/src/main.rs

```diff
@@ -52,6 +54,225 @@ pub fn main() {
         }
     };
 
+    let regtest = matches.is_present("regtest");
+    if regtest {
+        use std::ffi::OsString;
+        use std::fs::File;
+        use std::io;
+        use std::path::PathBuf;
+        use std::process::{Command, Stdio};
+        use std::{thread, time};
+
+        // confirm we are in a git directory, get info on it
+        let revparse_raw = Command::new("git")
+            .args(["rev-parse", "--show-toplevel"])
+            .output()
+            .expect("no git!? time to quit.");
+
+        let revparse = std::str::from_utf8(&revparse_raw.stdout).expect("revparse error");
```

Comment:
So, is `revparse` the `repo_root_name`?  How about that as a name for it?

It seems like `...rev-parse --show-toplevel...` is _how_ it was obtained and `repo_root_name` is _what_ it is.

How about rolling the entire derivation of that variable extraction into one expression?  Like:

`let repo_root_name = std::str::from_utf8(&Command::new("git").args(...`

Does that run afoul of ownership somewhere?